File: Backend/Entity_Layer/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entity_Layer
{
    public class Comment : Parent
    {
        [Required]
        [MinLength(1)]
        public string Body { get; set; }
        public long PostId { get; set; }
        public Post Post { get; set; }
        public IEnumerable<Reply> Replies { get; set; }
    }
}

File: Frontend/src/app/models/comment.ts
import { Post } from "./Post";
import { Reply } from "./reply";
import { User } from "./user";

export class Comment_
{
    constructor
    (
        public id: number,
        public body: string,
        public dateCreated: Date,
        public postId: number,
        public userId: string,
        public replies: Reply[],
        public post?: Post,
        public user?: User
    )
    {}

    static newComment(body: string, postId: number, replies?: Reply[]) {
        return new Comment_(0, body, new Date(), postId, '', replies || []);
    }
}